tests/system.rs

```diff
@@ -3045,4 +3055,151 @@ fn closure_bracket_scope_nested_match_outside() {
     let got = get_definition(src, None);
     assert_eq!("x", got.matchstr);
     assert_eq!("| x: i32 |", got.contextstr);
+}
+
+#[test]
+fn get_type_finds_explicit_local_var() {
+    let _lock = sync!();
+    let src = "
+    fn main() {
+        let value: Option<u16> = Some(5);
+        if val~ue.is_some() {
+            // do nothing
+        }
+    }
+    ";
+
+    let got = get_type(src, None);
+    assert_eq!("Option", got.matchstr);
```

Comment:
We don't put generic types into the `matchstr` property today; they're in the `generic_types` array instead. I could try and do something to push that data into `matchstr` but I'm not sure which approach would be better for tooling authors.